git/config.go

```diff
@@ -239,5 +243,68 @@ func ParseConfig(configFile io.Reader) GitConfig {
 			sections = append(sections, *section)
 		}
 	}
-	return GitConfig{sections}
+	return GitConfig{sections: sections}
+}
+
+func LoadGlobalConfig() (GitConfig, error) {
+	home := os.Getenv("HOME")
+	if home == "" {
+		home = os.Getenv("home") // On some OSes, it is home
+	}
+
+	if home == "" {
+		return GitConfig{}, fmt.Errorf("Global git configuration could not be found since HOME and home environment variables were not defined.")
+	}
+
+	fname := home + "/.gitconfig"
+
+	file, err := os.OpenFile(fname, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return GitConfig{}, err
+	}
+	config := ParseConfig(file)
+	err = file.Close()
+	if err != nil {
+		return GitConfig{}, err
+	}
+
+	config.fname = fname
+
+	return config, nil
+}
+
+func LoadLocalConfig(c *Client) (GitConfig, error) {
+	fname := filepath.Join(c.GitDir.String(), "config")
+
+	file, err := os.OpenFile(fname, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return GitConfig{}, err
+	}
+	config := ParseConfig(file)
+	err = file.Close()
+	if err != nil {
+		return GitConfig{}, err
+	}
+
+	config.fname = fname
+
+	return config, nil
+}
+
+func (g GitConfig) WriteConfig() error {
+	err := os.Remove(g.fname)
+	if err != nil {
+		return err
+	}
+	configFile, err := os.Create(g.fname)
+	if err != nil {
+		return err
+	}
+	g.WriteFile(configFile)
+	err = configFile.Close()
```

Comment:
This could just be "return configFile.Close()".

Reply:
👍